src/core/core.datasetController.js

```diff
@@ -379,18 +375,43 @@ export default class DatasetController {
 	 */
 	configure() {
 		const me = this;
-		me._config = merge(Object.create(null), [
-			defaults.controllers[me._type].datasets,
-			(me.chart.options[me._type] || {}).datasets,
-			me.getDataset(),
-		], {
-			merger(key, target, source) {
-				if (key !== 'data') {
-					_merger(key, target, source);
-				}
-			}
-		});
-		me._parsing = resolve([me._config.parsing, me.chart.options.parsing, true]);
+		me.options = me.resolveOptions();
+
+		me._parsing = valueOrDefault(me.options.parsing, true);
+	}
+
+	_mergeOptions(prefix) {
+		const me = this;
+		const config = me.chart.config;
+		const datasetType = me._type;
+		const scopes = [`controllers.${datasetType}.datasets`, `${datasetType}.datasets`];
+		const prefixes = prefix ? [prefix] : [];
+		if (me.datasetElementType) {
```

Comment:
Does this mean line options are considered before point options? Should they be mutually exclusive?

Reply:
It does. These end up in the "controller options" (this.options), and ideally we would not have any element options there.
So good names for these functions is another thing to refactor still. I think this should be called `_mergeControllerOptions`.

Reply:
I like `_mergeControllerOptions`